ocaml/xapi/xapi.ml

```diff
@@ -675,8 +703,29 @@ let check_network_reset () =
               | None ->
                   phy_pif
             in
-            Xapi_pif.reconfigure_ip ~__context ~self:pif ~mode ~iP ~netmask
-              ~gateway ~dNS ;
+            ( match mode with
+            | Some mode ->
+                Xapi_pif.reconfigure_ip ~__context ~self:pif ~mode ~iP ~netmask
+                  ~gateway ~dNS
+            | None ->
+                ()
+            ) ;
```

Comment:
`Option.iter` is useful here: 
```suggestion
            Option.iter (fun mode ->
                Xapi_pif.reconfigure_ip ~__context ~self:pif ~mode ~iP ~netmask
                  ~gateway ~dNS)
              mode ;
```

Reply:
Didn't know about that, thanks it's usefull indeed!